src/Optimizer/DefaultConfiguration.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+namespace AmpProject\Optimizer;
+
+use AmpProject\Optimizer\Configuration\AmpRuntimeCssConfiguration;
+use AmpProject\Optimizer\Configuration\PreloadHeroImageConfiguration;
+use AmpProject\Optimizer\Configuration\RewriteAmpUrlsConfiguration;
+use AmpProject\Optimizer\Configuration\TransformedIdentifierConfiguration;
+use AmpProject\Optimizer\Exception\InvalidConfigurationValue;
+use AmpProject\Optimizer\Exception\UnknownConfigurationClass;
+use AmpProject\Optimizer\Exception\UnknownConfigurationKey;
+use AmpProject\Optimizer\Transformer\AmpRuntimeCss;
+use AmpProject\Optimizer\Transformer\PreloadHeroImage;
+use AmpProject\Optimizer\Transformer\RewriteAmpUrls;
+use AmpProject\Optimizer\Transformer\TransformedIdentifier;
+
+/**
+ * Configuration object that validates and stores configuration settings.
+ *
+ * @package ampproject/amp-toolbox
+ */
+class DefaultConfiguration implements Configuration
+{
+
+    /**
+     * Associative array of already validated configuration settings.
+     *
+     * @var array
+     */
+    protected $configuration;
+
+    /**
+     * Associative array mapping the transformer classes to their configuration classes.
+     *
+     * This can be extended by third-parties via:
+     *
+     * @see registerConfigurationClass()
+     *
+     * @var array
+     */
+    protected $transformerConfigurationClasses = [
+        AmpRuntimeCss::class         => AmpRuntimeCssConfiguration::class,
+        PreloadHeroImage::class      => PreloadHeroImageConfiguration::class,
+        RewriteAmpUrls::class        => RewriteAmpUrlsConfiguration::class,
+        TransformedIdentifier::class => TransformedIdentifierConfiguration::class,
+    ];
+
+    /**
+     * Instantiate a Configuration object.
+     *
+     * @param array $configurationData Optional. Associative array of configuration data to use. This will be merged
+     *                                 with the default configuration and take precedence.
+     */
+    public function __construct($configurationData = [])
+    {
+        $this->configuration = array_merge(
+            Configuration::DEFAULTS,
```

Comment:
Would this be better so it can be overridden by child classes?

```suggestion
            static::DEFAULTS,
```